test(dialect): Update GaussDB dialect configuration and tests

SweetWuXiaoMei opened the pull request:
- Modified the database connection information of GaussDB - Adjusted the version compatibility of the GaussDBDialect class - Updated multiple test cases to adapt to the characteristics of GaussDB -Optimized some SQL statements to comply with the syntax requirements of GaussDB

SweetWuXiaoMei committed:
test(dialect): Update GaussDB dialect configuration and tests

- Modified the database connection information of GaussDB - Adjusted the version compatibility of the GaussDBDialect class - Updated multiple test cases to adapt to the characteristics of GaussDB
-Optimized some SQL statements to comply with the syntax requirements of GaussDB

liubao68 commented on hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/QueryTest.java:
```diff
@@ -600,17 +605,22 @@ public void testNativeQueryContainsQuotedSemicolonWithLimit() {
+//			assertEquals( 1, q.getResultList().size() );
+
+//			q = em.createNativeQuery( "select * from Item where name like '%;%' ; " ).setMaxResults(10);
+//			assertEquals( 1, q.getResultList().size() );
```
Add `TODO` for this test case, because will be fixed later.

liubao68 commented on hibernate-core/src/test/java/org/hibernate/orm/test/jpa/query/QueryTest.java:
```diff
@@ -600,17 +605,22 @@ public void testNativeQueryContainsQuotedSemicolonWithLimit() {
 	@SkipForDialect(value = SybaseDialect.class, comment = "Doesn't support semicolon as ending of statement")
 	@SkipForDialect(value = DerbyDialect.class, comment = "Doesn't support semicolon as ending of statement")
 	@SkipForDialect(value = DB2Dialect.class, comment = "Doesn't support semicolon as ending of statement")
+//	@SkipForDialect(value = GaussDBDialect.class, comment = "Doesn't support semicolon as ending of statement")
```
Delete this comment

liubao68 commented on gradle/databases.gradle:
```diff
@@ -124,8 +124,8 @@ ext {
-                        'jdbc.user'  : 'hibernateormtest',
-                        'jdbc.pass'  : 'hibernateormtest',
+                        'jdbc.user'  : 'root',
+                        'jdbc.pass'  : 'chenzhida',
```
Create a new user `hibernate_orm_test` with password `hibernateormtest`.   Or do not merge this file.

liubao68 replied:
do not commit this file

SweetWuXiaoMei committed:
test(dialect): Skip GaussDB tests that do not support array_position function

- Skip GaussDB tests in multiple unit tests- Update the registration logic of GaussDB related functions in CommonFunctionFactory- Add a GaussDB-specific array operation function class
- Adjust gradle configuration and update GaussDB connection parameters- Add checks for GaussDB in DialectFeatureChecks

liubao68 commented on gradle.properties:
```diff
@@ -1,7 +1,7 @@
 # We set '-Dlog4j2.disableJmx=true' to prevent classloader leaks triggered by the logger.
 # (Some of these settings need to be repeated in the test.jvmArgs blocks of each module)
-org.gradle.jvmargs=-Dlog4j2.disableJmx -Xmx2g -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Duser.language=en -Duser.country=US -Duser.timezone=UTC -Dfile.encoding=UTF-8
+org.gradle.jvmargs=-Dlog4j2.disableJmx -Xmx4g -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Duser.language=en -Duser.country=US -Duser.timezone=UTC -Dfile.encoding=UTF-8
```
Do not commit this fix.

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/GaussDBDialect.java:
```diff
@@ -557,21 +561,21 @@ public void initializeFunctionRegistry(FunctionContributions functionContributio
 		functionFactory.arrayAggregate();
-		functionFactory.arrayPosition_gaussdb();
-		functionFactory.arrayPositions_gaussdb();
+//		functionFactory.arrayPosition_gaussdb();
+//		functionFactory.arrayPositions_gaussdb();
```
delete this code and function `arrayPosition_gaussdb`

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/GaussDBDialect.java:
```diff
@@ -588,7 +592,7 @@ public void initializeFunctionRegistry(FunctionContributions functionContributio
 		functionFactory.jsonInsert_gaussdb();
 		functionFactory.jsonMergepatch_gaussdb();
-		functionFactory.jsonArrayAppend_gaussdb(true);
+//		functionFactory.jsonArrayAppend_gaussdb(true);
```
delete this code and `jsonArrayAppend_gaussdb`

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/GaussDBDialect.java:
```diff
@@ -1327,11 +1331,10 @@ public boolean supportsRecursiveCTE() {
-			case PERCENT_ONLY, PERCENT_WITH_TIES -> false;
-			case ROWS_WITH_TIES -> getVersion().isSameOrAfter(13);
-		};
+//		if ( type.equals( FetchClauseType.ROWS_ONLY ) ) {
+//			return true;
+//		}
```
delete this code

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java:
```diff
@@ -2852,10 +2852,11 @@ public void arrayPosition_postgresql() {
-	public void arrayPosition_gaussdb() {
-		functionRegistry.register( "array_position", new GaussDBArrayPositionFunction( typeConfiguration ) );
-	}
+//	public void arrayPosition_gaussdb() {
+//		functionRegistry.register( "array_position", new GaussDBArrayPositionFunction( typeConfiguration ) );
+//	}
```
delete this code

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/function/CommonFunctionFactory.java:
```diff
@@ -2894,17 +2895,18 @@ public void arrayPositions_postgresql() {
-				new GaussDBArrayPositionsFunction( true, typeConfiguration )
-		);
-	}
+//	public void arrayPositions_gaussdb() {
+//		functionRegistry.register(
+//				"array_positions",
+//				new GaussDBArrayPositionsFunction( false, typeConfiguration )
+//		);
+//		functionRegistry.register(
+//				"array_positions_list",
+//				new GaussDBArrayPositionsFunction( true, typeConfiguration )
+//		);
+//	}
```
delete this code

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/function/array/GaussDBArrayRemoveFunction.java:
```diff
@@ -0,0 +1,63 @@
+import java.util.List;
+
+/**
+ * HSQLDB array_remove function.
```
comments use GaussDB

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/function/array/GaussDBArrayRemoveIndexFunction.java:
```diff
@@ -0,0 +1,67 @@
+import java.util.List;
+
+/**
+ * HSQLDB array_remove function.
```
comments use GaussDB

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/function/array/GaussDBArrayReplaceFunction.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.List;
+
+/**
+ * Oracle array_replace function.
```
comments use GaussDB

liubao68 commented on hibernate-core/src/main/java/org/hibernate/dialect/function/array/GaussDBArrayTrimFunction.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.List;
+
+/**
+ * Oracle array_trim function.
```
comments use GaussDB

liubao68 commented on hibernate-core/src/test/java/org/hibernate/orm/test/annotations/collectionelement/ElementCollectionOfEmbeddableWithEntityWithEntityCollectionTest.java:
```diff
@@ -4,35 +4,35 @@
-import java.util.ArrayList;
-import java.util.List;
-
+import jakarta.persistence.CollectionTable;
+import jakarta.persistence.ElementCollection;
+import jakarta.persistence.Embeddable;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OrderColumn;
+import jakarta.persistence.Table;
+import org.assertj.core.api.Assertions;
 import org.hibernate.Hibernate;
 import org.hibernate.annotations.Cascade;
 import org.hibernate.annotations.Fetch;
+import org.hibernate.dialect.GaussDBDialect;
 import org.hibernate.orm.test.annotations.collectionelement.ElementCollectionOfEmbeddableWithEntityWithEntityCollectionTest.Event;
```
Do you use code template of hibernate?

SweetWuXiaoMei committed:
refactor(dialect): refactor the GaussDB dialect and update related configurations - Removed unsupported GaussDB function implementations
- Updated GaussDB database connection configurations
- Optimized Gradle build configurations
- Adjusted JVM parameter settings

SweetWuXiaoMei committed:
refactor(dialect): Optimize the code structure and comments of GaussDB array functions

- Modify the comments of the GaussDBArrayRemoveIndexFunction class to clarify its function
- Add comments to the GaussDBArraySetFunction class to explain its function
- Update the comments of the GaussDBArrayTrimFunction class to clarify its function and add code source description
